src/main/java/net/fabricmc/loader/transformer/accesswidener/AccessWidener.java

```diff
@@ -0,0 +1,359 @@
+/*
+ * Copyright 2016 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.loader.transformer.accesswidener;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.LinkedHashSet;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.IntUnaryOperator;
+
+import org.objectweb.asm.Opcodes;
+
+import net.fabricmc.loader.FabricLoader;
+import net.fabricmc.loader.api.ModContainer;
+import net.fabricmc.loader.metadata.LoaderModMetadata;
+import net.fabricmc.mappings.EntryTriple;
+
+public class AccessWidener {
+	public String namespace;
+	public Map<String, Access> classAccess = new HashMap<>();
+	public Map<EntryTriple, Access> methodAccess = new HashMap<>();
+	public Map<EntryTriple, Access> fieldAccess = new HashMap<>();
+	private Set<String> classes = new LinkedHashSet<>();
+
+	private final FabricLoader fabricLoader;
+
+	public AccessWidener(FabricLoader fabricLoader) {
+		this.fabricLoader = fabricLoader;
+	}
+
+	public void loadFromMods() {
+		for (ModContainer modContainer : fabricLoader.getAllMods()) {
+			LoaderModMetadata modMetadata = (LoaderModMetadata) modContainer.getMetadata();
+			String accessWidener = modMetadata.getAccessWidener();
+
+			if (accessWidener != null) {
+				Path path = modContainer.getPath(accessWidener);
+
+				try (BufferedReader reader = Files.newBufferedReader(path)) {
+					read(reader, fabricLoader.getMappingResolver().getCurrentRuntimeNamespace());
+				} catch (Exception e) {
+					throw new RuntimeException("Failed to read accessWidener file from mod " + modMetadata.getId(), e);
+				}
+			}
+		}
+	}
+
+	public void read(BufferedReader reader, String currentNamespace) throws IOException {
+		String[] header = reader.readLine().split("\\s+");
+
+		if (header.length != 3 || !header[0].equals("accessWidener")) {
+			throw new UnsupportedOperationException("Invalid access access widener file");
+		}
+
+		if (!header[1].equals("v1")) {
+			throw new RuntimeException(String.format("Unsupported access widener format (%s)", header[1]));
+		}
+
+		if (!header[2].equals(currentNamespace)) {
```

Comment:
Can we just do dynamic namespace remapping instead, which the mapping parser already allows? We cannot always assume that we run in intermediary.

Reply:
The namespace matches the same as the mod's classes. When we do runtime remapping of mods then this can be done.

I dont see a reason for it now.